magefile.go

```diff
@@ -3503,71 +3487,43 @@ func (Helm) RenderExamples() error {
 	return nil
 }
 
+// UpdateAgentVersion updates the agent version in the Elastic-Agent and EDOT-Collector Helm charts.
 func (Helm) UpdateAgentVersion() error {
-	for _, chart := range helmChartsValues {
-		valuesFile := filepath.Join(chart.path, "values.yaml")
-
-		data, err := os.ReadFile(valuesFile)
-		if err != nil {
-			return fmt.Errorf("failed to read file: %w", err)
-		}
-
-		isTagged, err := devtools.TagContainsCommit()
-		if err != nil {
-			return fmt.Errorf("failed to check if tag contains commit: %w", err)
-		}
-
-		if !isTagged {
-			isTagged = os.Getenv(snapshotEnv) != ""
-		}
-
-		agentVersion := getVersion()
-
-		// Parse YAML into a Node structure because
-		// it maintains comments
-		var rootNode yaml.Node
-		err = yaml.Unmarshal(data, &rootNode)
-		if err != nil {
-			return fmt.Errorf("failed to unmarshal YAML: %w", err)
-		}
-
-		if rootNode.Kind != yaml.DocumentNode {
-			return fmt.Errorf("root node is not a document node")
-		} else if len(rootNode.Content) == 0 {
-			return fmt.Errorf("root node has no content")
-		}
-
-		if err := updateYamlNodes(rootNode.Content[0], agentVersion, chart.versionKeys...); err != nil {
+	agentVersion := bversion.GetParsedAgentPackageVersion().CoreVersion()
+	agentSnapshotVersion := agentVersion + "-SNAPSHOT"
+	// until the Helm chart reaches GA this remains with -beta suffix
+	agentChartVersion := agentVersion + "-beta"
```

Comment:
I *think* what we want to do here is something like: If the agent version has the `-SNAPSHOT` suffix, then Chart version should be the same as agent version. Otherwise, Chart version should be agent version + "beta". Does that make sense?

Reply:
That sounds good to me. We can do that in a separate PR though, this one looks good to me. Thanks for being patient with my feedback, in hindsight I probably should've brought this up in the issue.